Shell/main.cpp

```diff
@@ -379,23 +410,188 @@ static int sh_time(int argc, const char** argv)
     return exit_code.value();
 }
 
+static int sh_jobs(int argc, const char** argv)
+{
+    bool list = false, show_pid = false;
+
+    Core::ArgsParser parser;
+    parser.add_option(list, "List all information about jobs", "list", 'l');
+    parser.add_option(show_pid, "Display the PID of the jobs", "pid", 'p');
+
+    if (!parser.parse(argc, const_cast<char**>(argv), false))
+        return 1;
+
+    enum {
+        Basic,
+        OnlyPID,
+        ListAll,
+    } mode { Basic };
+
+    if (show_pid)
+        mode = OnlyPID;
+
+    if (list)
+        mode = ListAll;
+
+    for (auto& job : g.jobs) {
+        auto pid = job.value.pid();
+        int wstatus;
+        auto rc = waitpid(pid, &wstatus, WNOHANG);
+        if (rc == -1) {
+            perror("waitpid");
+            return 1;
+        }
+        auto status = "running";
+
+        if (rc != 0) {
+            if (WIFEXITED(wstatus))
+                status = "exited";
+
+            if (WIFSTOPPED(wstatus))
+                status = "stopped";
+
+            if (WIFSIGNALED(wstatus))
+                status = "signaled";
+        }
+
+        switch (mode) {
+        case Basic:
+            printf("[%llu] %s %s\n", job.value.job_id(), status, job.value.cmd().characters());
+            break;
+        case OnlyPID:
+            printf("[%llu] %d %s %s\n", job.value.job_id(), pid, status, job.value.cmd().characters());
+            break;
+        case ListAll:
+            printf("[%llu] %d %d %s %s\n", job.value.job_id(), pid, job.value.pgid(), status, job.value.cmd().characters());
+            break;
+        }
```

Comment:
Idea for the future: could we make the jobs in this list be links to open them in SystemManager, somehow?

Reply:
Could certainly be an interesting feature.
Perhaps a unique protocol for launchserver?